app/legacy_lib/merge_supporters.rb

```diff
@@ -104,10 +104,10 @@ def self.merge_by_id_groups(np_id, arr_of_ids, profile_id, skip_conflicting_cust
 
         result = MergeSupporters.selected(data, ids, np_id, profile_id, skip_conflicting_custom_fields)
         supporter_ids_with_conflicting_custom_fields << ids if result[:status] == :failure
+
+        Supporter.find_by(name: data['name']).publish_created
```

Comment:
This won't work for a few reasons:

* supporters aren't scoped to the nonprofit
* 'name' is not necessarily unique, after all, multiple people can have the same name
* This search will include deleted supporters

In effect this is calling publish create on the first supporter the database finds, either deleted or not, among all nonprofits, which happens to have the same name as the one we're saving.

You should be calling on the specific supporter being created by MergeSupporters.selected. The easiest way to do that is to get one of the supporters in ids and look for the merged_into Supporter record. That's the one that should have the publish_created done on.